File: Plugins/Aspose_Imaging_Java_for_Ruby/lib/asposeimagingjava/images/convertingrasterimages.rb
module Asposeimagingjava
  module ConvertingRasterImages
    def initialize()
        # Binarization with Fixed Threshold
        binarization_with_fixed_threshold()

        # Binarization with Otsu Threshold
        binarization_with_otsu_threshold()

        # Transform image to its grayscale representation
        transform_image_to_grayscale()
    end

    def binarization_with_fixed_threshold()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Check if image is cached
        if !image.isCached()
            # Cache image if not already cached
            image.cacheData()
        end

        # Binarize image with predefined fixed threshold
        image.binarizeFixed(100)

        # Save the image to disk
        image.save(data_dir + "binarization_with_fixed_threshold.jpg")

        # Display Status.
        puts "Binarization image with Fixed Threshold successfully!"
    end

    def binarization_with_otsu_threshold()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Check if image is cached
        if !image.isCached()
            # Cache image if not already cached
            image.cacheData()
        end

        # Binarize image with Otsu Thresholding
        image.binarizeOtsu()

        # Save the image to disk
        image.save(data_dir + "binarization_with_otsu_threshold.jpg")

        # Display Status.
        puts "Binarization image with Otsu Threshold successfully!"
    end

    def transform_image_to_grayscale()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Check if image is cached
        if !image.isCached()
            # Cache image if not already cached
            image.cacheData()
        end

        # Transform image to its grayscale representation
        image.grayscale()

        # Save the image to disk
        image.save(data_dir + "transform_image_to_grayscale.jpg")

        # Display Status.
        puts "Transform image to its grayscale representation successfully!"
    end
  end
end

File: Plugins/Aspose_Imaging_Java_for_PHP/src/aspose/imaging/ManagingRasterFormats/ConvertingRasterImages.php
<?php
namespace Aspose\Imaging\ManagingRasterFormats;

use com\aspose\imaging\Image as Image;
class ConvertingRasterImages{

    public static function run($dataDir=null){

        # Binarization with Fixed Threshold
        ConvertingRasterImages::binarization_with_fixed_threshold($dataDir);

        # Binarization with Otsu Threshold
        ConvertingRasterImages::binarization_with_otsu_threshold($dataDir);

        # Transform image to its grayscale representation
        ConvertingRasterImages::transform_image_to_grayscale($dataDir);
    }

    public static function binarization_with_fixed_threshold($dataDir=null){

        $image=new Image();
        $image = $image->load($dataDir."test.jpg");


        # Check if image is cached
        if (!$image->isCached()) {
        # Cache image if not already cached
        $image->cacheData();
        }

        # Binarize image with predefined fixed threshold
        $image->binarizeFixed(100);

        # Save the image to disk
        $image->save($dataDir . "binarization_with_fixed_threshold.jpg");

        # Display Status.
        print "Binarization image with Fixed Threshold successfully!".PHP_EOL;
    }

    public static function binarization_with_otsu_threshold($dataDir=null)
    {

        # Load an existing image
        $image = new Image();
        $image = $image->load($dataDir."test.jpg");

        # Check if image is cached
        if (!$image->isCached()) {
        # Cache image if not already cached
        $image->cacheData();
        }

        # Binarize image with Otsu Thresholding
        $image->binarizeOtsu();

        # Save the image to disk
        $image->save($dataDir."binarization_with_otsu_threshold.jpg");

        # Display Status.
        print "Binarization image with Otsu Threshold successfully!".PHP_EOL;
    }

    public static function transform_image_to_grayscale($dataDir=null){

        # Load an existing image
        $image=new Image();
        $image = $image->load($dataDir."test.jpg");

        # Check if image is cached
        if (!$image->isCached()) {
        # Cache image if not already cached
        $image->cacheData();
        }

        # Transform image to its grayscale representation
        $image->grayscale();

        # Save the image to disk
        $image->save($dataDir."transform_image_to_grayscale.jpg");

        # Display Status.
        print "Transform image to its grayscale representation successfully!".PHP_EOL;
    }
}
?>